File: VTV30012.java
import java.util.Scanner;

public class Main {

    static String[] l = new String[1000000];
    static int n = 1;
    

    public static String solve(int i) {
        if (i >= n) {
            return "Yes'";
        }
        if (Float.parseFloat(l[i]) < 0) {
            return solve(i + 1);
        } else {
            return "No";
        }
    }

    public static void output(String n) {
        System.out.println(n);
    }

    public static void input() {
     Scanner scanner = new Scanner(System.in);
        n = Integer.parseInt(scanner.nextLine());
        l = scanner.nextLine().trim().split(" ");
    }

    public static void main(String[] args) {
       
Scanner scanner = new Scanner(System.in);
        int testcase = Integer.parseInt(scanner.nextLine());
        
        for (int i = 0; i < testcase; i++) {
            String n = solve(0);
            output(n);
            
        }
        
    }
}

File: VTV30012.dart
import 'dart:io';

List<String> l = [];
int n = 1;

input() {
  n = int.parse(stdin.readLineSync() ?? '');
  l = (stdin.readLineSync() ?? '').split(' ');
}

String solve(int i) {
  if (i >= n) {
    return 'Yes';
  }
  if (double.parse(l[i]) < 0) {
    return solve(i + 1);
  } else {
    return 'No';
  }
}

void output(String n) {
  print(n);
}

void main() {
  int testcase = int.parse(stdin.readLineSync() ?? '');
  for (int i = 0; i < testcase; i++) {
    input();
    String a = solve(0);
    output(a);
  }
}
